test/e2e/util.go

```diff
@@ -337,19 +337,27 @@ func registryPodHealthy(address string) bool {
 	return true
 }
 
-func catalogSourceRegistryPodSynced(catalog *operatorsv1alpha1.CatalogSource) bool {
-	registry := catalog.Status.RegistryServiceStatus
-	connState := catalog.Status.GRPCConnectionState
-	if registry != nil && connState != nil && !connState.LastConnectTime.IsZero() && connState.LastObservedState == "READY" {
-		fmt.Printf("probing catalog %s pod with address %s\n", catalog.GetName(), registry.Address())
-		return registryPodHealthy(registry.Address())
-	}
-	state := "NO_CONNECTION"
-	if connState != nil {
-		state = connState.LastObservedState
+func catalogSourceRegistryPodSynced() func(catalog *operatorsv1alpha1.CatalogSource) bool {
+	var lastState string
+	lastTime := time.Now()
```

Comment:
Interesting... (more of a "oh, wait, golang let's you do some interesting stuff")